src/MPI_with_SYCL/CMakeLists.txt

```diff
@@ -1,11 +1,69 @@
+function(run_mpi_test)
+    cmake_parse_arguments(RUN_MPI_TEST "" "BINARY;BACKEND_SELECTOR" "" ${ARGN})
+    execute_process(
+        COMMAND bash -c "! ONEAPI_DEVICE_SELECTOR=${RUN_MPI_TEST_BACKEND_SELECTOR}:gpu mpirun -n 2 ${RUN_MPI_TEST_BINARY}"
+        RESULT_VARIABLE MPI_TEST_SUCCEEDED
+        OUTPUT_QUIET
+        ERROR_QUIET)
+    set(MPI_TEST_SUCCEEDED ${MPI_TEST_SUCCEEDED} PARENT_SCOPE)
+endfunction()
+
+function(test_mpi_gpu_support)
+    set(MPI_TEST_COMPILE_ARGS ${SYCL_FLAGS})
+    list(APPEND MPI_TEST_COMPILE_ARGS ${MPI_CXX_COMPILE_OPTIONS} -I${MPI_CXX_INCLUDE_DIRS} ${MPI_CXX_LINK_FLAGS} ${MPI_LIBRARIES})
+    list(JOIN MPI_TEST_COMPILE_ARGS " " MPI_TEST_COMPILE_ARGS)
+    set(MPI_TEST_BINARY ${CMAKE_CURRENT_BINARY_DIR}/test-mpi-gpu)
+    set(MPI_TEST_COMPILE_CMD "${CMAKE_CXX_COMPILER} ${MPI_TEST_COMPILE_ARGS} -o ${MPI_TEST_BINARY} ${CMAKE_CURRENT_SOURCE_DIR}/send_recv_usm.cpp")
+    execute_process(
```

Comment:
You might be able to use the CMake `try_compile` here, as it has a variable which is the output of the compilation process. As we discussed there doesn't seem to be a way to use the mpirun launcher with `try_run` though so the run_mpi_test function will likely continue to be required.

Reply:
Thanks, that sounds like a bit of a pain - so thank you for putting the effort in :)